rootToH5converter/Jet.py

```diff
@@ -167,21 +167,26 @@ def add_constituents(self, physObjects, pt_cut, delta_r, iJet=-1, track_jet_inde
                     if delta_eta ** 2. + delta_phi ** 2. < delta_r ** 2.:
                         constituents.append(vec)
         else:
-            for i, track in enumerate(physObjects):
-                jet_index = track_jet_index[i]
-                if jet_index == iJet:
-                    constituents.append(track.get_four_vector())
+            if track_cand_index is None:
+                for i, track in enumerate(physObjects):
+                    jet_index = track_jet_index[i]
+                    if jet_index == iJet:
+                        constituents.append(track.get_four_vector())
+            else:
+                cand_indices_for_jet = (track_jet_index == iJet)
+                cand_indices = track_cand_index[cand_indices_for_jet]
+                constituents = [track.get_four_vector() for itrack, track in enumerate(physObjects) if itrack in cand_indices]
```

Comment:
Help me here: cand_indices_for_jet is a bool, which then you use to select indices from track_cand_index. Does it always have two elements?

Reply:
`cand_indices_for_jet` is a boolean jagged array that has the same jagged structure as `track_jet_index` (actually here I think it's not even jagged because `track_jet_index` is the track collection for 1 event, right?).
> Does it always have two elements?
Why do you mean? This is an "array at once" (not sure about the terminology) operation, that is more efficient than doing `for` loops, that's why I did it this way (it's how things are supposed to be done with awkward arrays if I understood the principle correctly). If `track_jet_index` is empty, then `cand_indices_for_jet` is empty and so does `cand_indices`.

Reply:
Ok, I'll need to go through these lines with a debugger to understand what's going on there :D Or maybe you'll explain that to me step by step at some point.

Reply:
For instance if
```
track_jet_index = [[0, 0, 0, 1, 1, 2], [0, 0, 1, 1, 1, 1, 2], []]
track_cand_index = [[0, 1, 2, 5, 8, 9], [0, 2, 3, 9, 11, 12, 13], []]
```
Then for `ijet` = 1
```
cand_indices_for_jet = [[False, False, False, True, True, False, False], [False, False, True, True, True, True False], []]
cand_indices = [[5, 8], [3, 9, 11, 12], []]
```
And here I think it's not even jagged e.g. it's of the form [0, 0, 0, 1, 1, 2], not [[...], [...], ...]